glnx-xattrs.c

```diff
@@ -87,40 +87,47 @@ read_xattr_name_array (const char *path,
   while (p < xattrs+len)
     {
       ssize_t bytes_read;
-      char *buf;
-      GBytes *bytes = NULL;
+      g_autofree char *buf = NULL;
+      g_autoptr(GBytes) bytes = NULL;
 
+    again:
       if (fd != -1)
         bytes_read = fgetxattr (fd, p, NULL, 0);
       else
         bytes_read = lgetxattr (path, p, NULL, 0);
       if (bytes_read < 0)
         {
+          if (errno == ENODATA)
+            continue;
```

Comment:
Ooh, yes.  You were absolutely right.  This was revealed when I actually wrote a test:

New patches in https://github.com/cgwalters/libglnx/commits/checked-listxattr

Unfortunately since this is closed it looks like github won't do notifications.